app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  layout 'new-layout'
+
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+      reset_session
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = t'flashes.danger.login_failure'
```

Comment:
code ruby em dùng nháy kép nha
```suggestion
      flash.now[:danger] = t "flashes.danger.login_failure"
```

Reply:
dạ vag ạ